test/passing/issue289.ml

```diff
@@ -0,0 +1,79 @@
+let foo =
+  let open Gql in
+  [ field "id" ~doc:"Toy ID." ~args:[] ~typ:(non_null guid)
+      ~resolve:(function _ctx -> x.id )
+  ; field "id" ~doc:"Toy ID." ~args:[] ~typppp ~resolve:(function _ctx ->
+        x.id )
+  ; field "id" ~doc:"Toy ID." ~args:[] ~typ:(non_null guid)
+      ~resolve:(function
+      | A -> x.id
```

Comment:
This needs indentation

Reply:
I'm not sure how to fix this one. @jberdine ?

Reply:
indentation is still weird here. Should we add another box ?

Reply:
This is an example of the case where less indentation is a bit worse, that I don't know how to address. If a box is added around `~resolve:(function...)`, then `~resolve:(function` won't be able to be placed on the previous line unless the whole function expression fits. Since this can only happen on the last arg, I think there is no potential for confusion wrt confusing the body of the function with code that follows, as the following code will be indented less. I think that this case looks a bit unexpected visually, but is not actually confusing.